### PR TITLE
AUT-817: Set up SAML SOAP Proxy on Fargate

### PR DESCRIPTION
**Pull request by adityapahuja**
This change sets up a new service called SAML SOAP Proxy Fargate. It will run on Fargate alongside with SAML SOAP Proxy running on EC2 instances.

Author: @adityapahuja

### CODE REVIEW

File: terraform/modules/hub/hub_saml_soap_proxy.tf [hunk 1]
```diff
@@ -120,30 +199,63 @@ resource "aws_iam_role_policy_attachment" "saml_soap_proxy_parameter_execution"
   policy_arn = aws_iam_policy.saml_soap_proxy_parameter_execution.arn
 }
 
+resource "aws_iam_role_policy_attachment" "saml_soap_proxy_fargate_parameter_execution" {
+  role       = "${var.deployment}-saml-soap-proxy-fargate-execution"
```
Comment (philandstuff) [inline terraform/modules/hub/hub_saml_soap_proxy.tf:203]:
```suggestion
  role       = module.saml_soap_proxy_fargate.execution_role_name
```

File: terraform/modules/hub/kms.tf [hunk 2]
```diff
@@ -161,6 +161,7 @@ data "aws_iam_policy_document" "saml_soap_proxy" {
       identifiers = [
         "arn:aws:iam::${data.aws_caller_identity.account.account_id}:root",
         "arn:aws:iam::${data.aws_caller_identity.account.account_id}:role/${var.deployment}-saml-soap-proxy-execution",
+        "arn:aws:iam::${data.aws_caller_identity.account.account_id}:role/${var.deployment}-saml-soap-proxy-fargate-execution",
```
Comment (philandstuff) [inline terraform/modules/hub/kms.tf:164]:
```suggestion
        "arn:aws:iam::${data.aws_caller_identity.account.account_id}:role/${module.saml_soap_proxy_fargate.execution_role_name}",
```

File: terraform/modules/hub/hub_saml_soap_proxy.tf (hunk 1 above)
Comment (adityapahuja) [reply terraform/modules/hub/hub_saml_soap_proxy.tf]: Done

File: terraform/modules/hub/kms.tf (hunk 2 above)
Comment (adityapahuja) [reply terraform/modules/hub/kms.tf]: Done

**Pull request merged**